### PR TITLE
Interactive controller refactoring and functional upgrade

### PR DESCRIPTION
**Pull request by shtukas**
## What does this change?

Another step in the interactives DCR migration. Follow up of: https://github.com/guardian/frontend/pull/23729

Here again we do not change much of what is already happening, if not posing the foundations for the `.json?dcr` functionality. To this aim we currently just return an empty JSON object. The rest is identical. 

Regular display : 

Screenshot 2021-04-22 at 12 06 12

Regular display with the currently inactive `?dcr` flag :

Screenshot 2021-04-22 at 12 07 18

`json` (legacy)

Screenshot 2021-04-22 at 12 19 43

`.json?dcr` : 

Screenshot 2021-04-22 at 12 27 47

And manually making sure the AMP election tracker is still there 🙂  (hardcoded AMP in the code for the screenshot) :

Screenshot 2021-04-22 at 12 09 17

### CODE REVIEW

**Review by nicl — APPROVED** [review]
Nice! Although I wonder if this PR is *too* small? Too big PRs are definitely hard to review, but too small can also make things difficult - as it is harder to get an overall sense of approach/change and review it holistically.

File: applications/app/controllers/InteractiveController.scala [hunk 1]
```diff
@@ -104,22 +105,25 @@ class InteractiveController(
...
-      case DotcomRendering => {
-        // On purpose reproduce Regular [work in progress]
-        lookup(path) map {
-          case Left(model)  => render(model)
-          case Right(other) => RenderOtherStatus(other)
-        }
-      }
+    val requestFormat = request.getRequestFormat
+    val renderingTier = ApplicationsInteractiveRendering.getRenderingTier(path)
+    println((requestFormat, renderingTier))
```
[the remark this answers is not in the document]
Comment (shtukas) [reply applications/app/controllers/InteractiveController.scala:118]: Yeah. I noticed that later and removed it.

File: applications/app/controllers/InteractiveController.scala [hunk 2]
```diff
@@ -104,22 +105,25 @@ class InteractiveController(
...
+    (requestFormat, renderingTier) match {
+      case (EmailFormat, _)                   => RenderItemLegacy(path: String)
```
[the remark this answers is not in the document]
Comment (shtukas) [reply applications/app/controllers/InteractiveController.scala:120]: Um... Yeah. At first I wanted to clarify to myself exactly what I was doing, but you are right.

**Comment by shtukas** [comment]
> Nice! Although I wonder if this PR is too small? 

@nicl 

I get you, but the two steps I made today are necessary regardless and I prefer going slow when modifying controllers.

To me, the important thing is to minimise the risk of each step, and to make each step logically contained, not the size of the change. 

My first step this morning was to handle the `?dcr` flag and the second step to pose the foundations for a specific `.json` object. Those two steps are the most delicate steps for me.  Now that the controller correctly isolate behaviour behind `.json?dcr` I know I can do wild stuff on that code path, and the rest I have planned can come in much larger chunks 🙂

**Pull request merged**